packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -253,29 +142,16 @@ const LoggedInPlansFeatureActionButton = ( {
 	const nonDefaultStorageOptionSelected = defaultStorageOption !== selectedStorageOptionForPlan;
 
 	if (
-		isFreePlan( planSlug ) ||
-		( storageAddOnsForPlan && ! canPurchaseStorageAddOns && nonDefaultStorageOptionSelected )
+		( isFreePlan( planSlug ) ||
+			( storageAddOnsForPlan && ! canPurchaseStorageAddOns && nonDefaultStorageOptionSelected ) ) &&
+		isP2FreePlan( planSlug ) &&
+		current
 	) {
-		if ( planActionOverrides?.loggedInFreePlan ) {
-			return (
-				<PlanButton planSlug={ planSlug } onClick={ onCtaClick } current={ current }>
-					{ planActionOverrides.loggedInFreePlan.text }
-				</PlanButton>
-			);
-		}
-
-		if ( isP2FreePlan( planSlug ) && current ) {
-			return null;
-		}
-
-		return (
-			<PlanButton planSlug={ planSlug } current={ current } disabled>
-				{ translate( 'Contact support', { context: 'verb' } ) }
-			</PlanButton>
-		);
+		return null;
```

Comment:
This could be the case where we return `{ primary: { callback: null, text: null } }` from `useAction`. So we can treat `null` returns as something that's meant to "show nothing", while `undefined` Action from `useAction` as something that's meant to show the defaults.

Reply:
I like your idea of having a null case and an undefined case. Makes sense to me.